clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -125,7 +160,7 @@ void cir::CIRDialect::initialize() {
 #define GET_OP_LIST
 #include "clang/CIR/Dialect/IR/CIROps.cpp.inc"
       >();
-  addInterfaces<CIROpAsmDialectInterface>();
+  addInterfaces<CIRInlinerInterface, CIROpAsmDialectInterface>();
```

Comment:
Incremental patches are way to go, I'm just a bit confused here: is this supposed to change any behavior at this point or is this just a skeleton that has no observed behavior? If this is changing behavior I'd expect a testcase and/or implementations for `isLegalToInline` to be returning false (for now).

Reply:
It does not change the behavior of CIR in this project since it is not used but allows other CIR users to use the inliner.
I was worried someone would ask for a test. :wink: An easy one would have been to just use the inliner pass (instead of the function) in `opt` except that a lot of other stuff is not yet implemented in CIR.

Reply:
I was thinking two types of tests:
- A test similar to `clang/test/CIR/Transforms/mem2reg.cir`.
- Better yet if we add a flag to cc1 as well, like we do for `mem2reg` in `clang/test/CIR/Transforms/mem2reg.c`
- Thoughts for the future: looks like we need a flag that enable all MLIR passes altogether, something like a `-fclangir-mlir-opt` of sorts, have you thought about that already?

Reply:
Yes, having a test with `--inline` in the same way as you suggest with `--mem2reg` but even with https://github.com/llvm/clangir/pull/1203 there are still some crashes with `--inline` because it requires an analysis to work to compute the benefit of doing the inlining.
I need to investigate this.
In my project I do not have the problem because I call the low-level API which does not compute any cost model and just do the job. :smile:

Reply:
Btw, it's fine if you initially add one small test that passes while other wild attempts to use might crash/ fail, as long as we don't enable this by default we should be good to incrementally improve it!